File: app/src/main/java/com/serwylo/lexica/lang/LanguageLabel.java
package com.serwylo.lexica.lang;

import android.content.Context;
import android.util.Log;

import androidx.annotation.NonNull;

public class LanguageLabel {

    private static final String TAG = "LanguageLabel";

    public static String getLabel(Context context, @NonNull Language language) {
        String prefName = "pref_dict_" + language.getName();
        int resId = context.getResources().getIdentifier(prefName, "string", context.getPackageName());
        if (resId <= 0) {
            Log.e(TAG, "Language " + language.getName() + " does not have a corresponding R.string.pref_dict_" + language.getName() + " value, defaulting to " + language.getName());
            return language.getName();
        }


        return context.getString(resId);
    }

}

File: app/src/main/java/com/serwylo/lexica/lang/LanguageLabel.kt
package com.serwylo.lexica.lang

import android.content.Context
import android.util.Log

object LanguageLabel {

    private const val TAG = "LanguageLabel"

    @JvmStatic fun getLabel(context: Context, language: Language): String {

        val prefName = "pref_dict_" + language.name

        val resId = context.resources.getIdentifier(prefName, "string", context.packageName)
        if (resId <= 0) {
            Log.e(TAG, "Language " + language.name + " does not have a corresponding R.string.pref_dict_" + language.name + " value, defaulting to " + language.name)
            return language.name
        }

        return context.getString(resId)

    }
}